packages/client-admin/src/pages/projectsettings.tsx

```diff
@@ -20,6 +19,66 @@ const IconPreview = (props: any) => {
   );
 };
 
+const ProjectAdditionalSettingsSwitch = () => {
+  const { decoded_token } = useAuth();
+  const projectId = decoded_token?.projectId || '';
+  const { data: projectData, called, loading } = useGetProjectQuery({ variables: { id: projectId }, skip: !projectId });
+  const [updateProjectAuthMethods] = useUpdateProjectAuthMethodsMutation();
+  const [updateProjectSettings] = useUpdateProjectSettingsMutation();
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    switch (event?.target.name) {
+      case 'displayProjectName':
+        try {
+          updateProjectSettings({ variables: { id: projectId, displayProjectName: event.target.checked } });
```

Comment:
NIT: Since each block is doing similar actions, it would be a bit cleaner to refactor to reduce the duplication. Something like.

```typescript
const params: <appropriate type> = { id: projectId };

switch (event?.target.name) {
  case 'displayProjectName':
    params.displayProjectName = event.target.checked;
    break;
  case: 'allowSignup':
    params.allowSignup = event.target.checked;
    break;
...
}

try {
  updateProjectSettings({ variables: params })
} catch (error) {
...
}
```